tap_hubspot/__init__.py

```diff
@@ -392,7 +395,7 @@ def get_v3_deals(v3_fields, v1_data):
     return v3_resp.json()['results']
 
 #pylint: disable=line-too-long
-def gen_request(STATE, tap_stream_id, url, params, path, more_key, offset_keys, offset_targets, v3_fields=None):
+def gen_request(STATE, tap_stream_id, url, params, path, more_key, offset_keys, offset_targets, v3_fields=None, write_state=True):
```

Comment:
alternative idea: instead of adding parameter, how about updating this method to handle if STATE is None? discover shouldn't have state so it can pass in None and this method can check STATE before getting/setting/writing it

Reply:
You're absolutely correct that we haven't had `STATE` during dicsovery before this change.
But as for your idea, I'm not sure it's going to work: before this change we didn't issue any requests to the server during discovery. Now we do, and `singer` manages offsets and paginated loading using `STATE` during contact lists load. We're going to need `STATE` to support it, esp. in case when we more than one page of contact lists available.

Reply:
i don't think the singer spec intended state to be used during discovery, i think it's meant to only be used during sync

Reply:
Got it now, thanks! Addressed.